src/imitation/algorithms/preference_comparisons.py

```diff
@@ -1316,6 +1319,34 @@ def _train(self, dataset: PreferenceDataset, epoch_multiplier: float = 1.0) -> N
             # sampler gives new indexes on every call
             bagging_dataset = data_th.Subset(dataset, list(sampler))
             self.member_trainers[member_idx]._train(bagging_dataset, epoch_multiplier)
+            keys = list(self.logger.name_to_value.keys())
+            for key in keys:
+                val = self.logger.name_to_value[key]
+                if key.startswith("mean/reward/epoch-"):
+                    new_key = key.replace(
+                        "mean/reward",
+                        f"mean/reward/member-{member_idx}",
+                    )
+                    self.logger.record(new_key, val)
```

Comment:
I'm confused by this code. It seems like we're recording things to a new key (based on the member index), but:
  1. Only the last value logged. (Maybe that's OK, if there's only a single timestep, that is `self.logger.dump` is never called inside `_train`?)
  2. We don't remove the old key.

Why not log with the right key in the first place? You could use the `add_prefix` method from Lev's PR https://github.com/HumanCompatibleAI/imitation/blob/4df85518203e5e83f34864d31ff807813f2a259c/src/imitation/util/logger.py (not yet merged, but you could always cherry-pick the logging commits into a separate PR, get that PR approved & merged, then go back and use it here). Alternatively you could follow the approach Rocamonde took with the regularization PR and pass around a prefix explicitly like in https://github.com/HumanCompatibleAI/imitation/blob/master/src/imitation/algorithms/preference_comparisons.py#L1223

Reply:
I have currently used Rocamonde's approach to log. Once #529 gets merged, we can switch the current code to using the `add_prefix` method in Lev's PR.